src/protocols/ssh/ssh.c

```diff
@@ -439,6 +439,7 @@ void* ssh_client_thread(void* data) {
         if (settings->server_alive_interval > 0) {
             timeout = 0;
             if (libssh2_keepalive_send(ssh_client->session->session, &timeout) > 0)
+                pthread_mutex_unlock(&(ssh_client->term_channel_lock));
                 break;
```

Comment:
You'll need to add the braces `{..}` for this `if` statement. By just adding this line the code will then always execute the `break` if the `settings->server_alive_interval` is greater than `0`, which is probably not what we want.

Reply:
Thanks, please check the new one